uint8 header
uint8 game_type
uint8 game_progress
uint16 stage_remain_time

uint16 red_1_hp
uint16 red_2_hp
uint16 red_3_hp
uint16 red_4_hp

uint16 red_5_hp
uint16 red_7_hp
uint16 blue_1_hp
uint16 blue_2_hp

uint16 blue_3_hp
uint16 blue_4_hp
uint16 blue_5_hp
uint16 blue_7_hp

uint16 red_base_hp
uint16 blue_base_hp

uint16 remain_hp
uint16 max_hp
uint16 max_power
uint16 heat0_limit
uint16 heat0
float32 power
float32 buffer
float32 bullet_speed
uint8 shooter_speed_limit

uint32 event_data